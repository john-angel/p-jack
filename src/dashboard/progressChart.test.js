import { TestScheduler } from 'jest';
import React from 'react';
import renderer from 'react-test-renderer';
import ProgressChart from './progressChart';

test('Progress chart UI', () => {
    const component = renderer.create(
        <ProgressChart tasksPercentage='20%' revenuePercentage='50%'></ProgressChart>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

test('35% on revenue/task completion use 10% of the progress bar', () => {
    let value = '35%';
    const component = new ProgressChart();
    expect(component.getValuePosition(value)).toBe('10%');   
})

test('65% on revenue/task completion use 40% of the progress bar', () => {
    const value = '65%';
    const component = new ProgressChart();
    expect(component.getValuePosition(value)).toBe('40%');
});

test('95% on revenue/task completion use 70% of the progress bar', () => {
    const value = '95%';
    const component = new ProgressChart();
    expect(component.getValuePosition(value)).toBe('70%');
})