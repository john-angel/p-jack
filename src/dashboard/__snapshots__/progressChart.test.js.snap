// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress chart UI 1`] = `
<section
  className="dashboardProgressChart"
>
  <p
    className="dashboardTasksNameProgressBar"
  >
    Tasks
  </p>
  <div
    className="dashboardTasksProgressBar"
    style={
      Object {
        "background": "linear-gradient(to right, #B290EB 0 20%, white 20% 100%)",
      }
    }
  >
    <p
      className="dashboardTasksValueProgressBar"
      style={
        Object {
          "transform": "translateX(10%)",
        }
      }
    >
      20%
    </p>
  </div>
  <p
    className="dashboardRevenueNameProgressBar"
  >
    Revenue
  </p>
  <div
    className="dashboardRevenueProgressBar"
    style={
      Object {
        "background": "linear-gradient(to right, #7ED3B2 0 50%, white 50% 100%)",
      }
    }
  >
    <div
      className="dashboardRevenueValueProgressBar"
      style={
        Object {
          "transform": "translateX(10%)",
        }
      }
    >
      50%
    </div>
  </div>
</section>
`;
